### PR TITLE
Initial stab at a simpler cocoadocs

### PR DESCRIPTION
**Pull request by orta**
I am planning on removing the docset generation from CocoaDocs. There is an entire blog post o the hows and the why, for now, I am taking a stab at this so that it can run at the same time as the current server, so I can be sure that things don't break. 

Oddly enough, CocoaDocs' current format of "one file to control them all" works well when making moves like this. This aims to re-use all the existing classes, but to do less. 

Not worth a code-review yet, my internet hasn't been good enough to actually try run through the full process.

### CODE REVIEW

File: classes/stats_generator.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@ def upload
       :dominant_language => @cloc_top[:language],
       :is_vendored_framework => is_vendored_framework(spec),
       :rendered_summary => spec.or_summary_html,
-      :builds_independently => supports_carthage,
+      :builds_independently =>  @test_carthage && supports_carthage,
```
Comment (segiddins) [inline classes/stats_generator.rb]: Extra space

**Comment by orta** [comment]
This CocoaPods side of it is good now, I talked with BuddyBuild who are interested in maintaining the CocoaDocs side, so I'm going to extend the scope of this PR to also include that. 

I can test this by running both scripts on CocoaDocs for a while to verify everything is still working in place correctly, then we can look at running the two scripts from different hosts.

File: cocoadocs-simple.rb [hunk 2]
```diff
@@ -3,29 +3,31 @@
...
 
-require "cocoapods-downloader"
-require "cocoapods-core"
-require "cocoapods"
+require 'cocoapods-downloader'
+require 'cocoapods-core'
+require 'cocoapods'
 
-require "open-uri"
+require 'open-uri'
```
Comment (segiddins) [inline cocoadocs-simple.rb]: open-uri is insecure, I'd advise strongly against using it

Comment (orta) [inline cocoadocs-simple.rb]: looks like it's being used to grab the podspec, will switch that out - thanks

**Comment by orta** [comment]
/cc @palleas

I'm going to start integrating this into CocoaDocs soon. It's at a place where it can be running on your infra. I'd recommend making an issue when you get time to start moving over. You want to verify that running both:

```sh
bundle exec ruby cocoadocs.rb cocoadocs url https://raw.githubusercontent.com/CocoaPods/Specs/master/Specs/6/b/f/ORStackView/3.0.1/ORStackView.podspec.json
```
and
```
bundle exec ruby cocoadocs.rb cocoadocs url https://raw.githubusercontent.com/CocoaPods/Specs/master/Specs/8/a/7/Moya/8.0.3/Moya.podspec.json
```

Once those work then you can start the Sinatra server with `bundle exec foreman start`.

**Comment by Palleas** [comment]
I've officially bribed of our devops into securing a first VM for CocoaDocs. This is happening!

**Comment by orta** [comment]
<3

**Comment by Palleas** [comment]
http://buddybuild-cocoadocs.s3-us-west-2.amazonaws.com/docsets/Moya/8.0.3/index.html

http://buddybuild-cocoadocs.s3-us-west-2.amazonaws.com/docsets/ORStackView/3.0.1/index.html

🎉

**Comment by dnkoutso** [comment]
This is cool stuff :)

**Comment by orta** [comment]
This is a great start! Looks like the Swift documentation isn't building right ( this is reasonable, given as it is exponentially more complex) 

http://buddybuild-cocoadocs.s3-us-west-2.amazonaws.com/docsets/ORStackView/3.0.1/index.html

**Comment by orta** [comment]
Hah - CocoaDocs dies on Moya 8.0.3 too - http://cocoadocs.org/docsets/Moya/8.0.3/

**Comment by orta** [comment]
I've got time this weekend to give this a shot

**Comment by Palleas** [comment]
Cool! I've been working on other stuff last week so no pressure.
I'm not super familiar with this codebase yet but if you think there are simple stuff that needs to be done, let me know :)

**Comment by orta** [comment]
I'd recommend following the repo, there's a request every day or two that I'm not really handling ATM. :P

**Comment by Palleas** [comment]
(Kidding, I'll take a look later today 😁 )

**Comment by orta** [comment]
Fixed all swift libs see: Moya 8.0.4 - http://cocoadocs.org/docsets/Moya/8.0.4/

**Comment by orta** [comment]
Deployed

**Comment by Palleas** [comment]
🎉

**Pull request merged**